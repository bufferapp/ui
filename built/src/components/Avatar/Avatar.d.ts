import React from 'react';
import PropTypes from 'prop-types';
declare const Avatar: {
    ({ src, alt, type, size, network, fallbackUrl }: {
        src: any;
        alt: any;
        type: any;
        size: any;
        network: any;
        fallbackUrl: any;
    }): React.JSX.Element;
    propTypes: {
        src: PropTypes.Validator<string>;
        alt: PropTypes.Validator<string>;
        fallbackUrl: PropTypes.Requireable<string>;
        type: PropTypes.Requireable<string>;
        size: PropTypes.Requireable<string>;
        network: PropTypes.Requireable<string>;
    };
    defaultProps: {
        type: string;
        size: string;
        fallbackUrl: string;
        network: null;
    };
};
export default Avatar;
//# sourceMappingURL=Avatar.d.ts.map