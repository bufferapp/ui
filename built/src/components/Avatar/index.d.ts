export { default } from './Avatar';
//# sourceMappingURL=index.d.ts.map