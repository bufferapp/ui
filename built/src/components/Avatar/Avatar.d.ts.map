{"version":3,"file":"Avatar.d.ts","sourceRoot":"","sources":["../../../../src/components/Avatar/Avatar.tsx"],"names":[],"mappings":"AACA,OAAO,KAAK,MAAM,OAAO,CAAA;AAEzB,OAAO,SAAS,MAAM,YAAY,CAAA;AAmJlC,QAAA,MAAM,MAAM;;;;;;;;;;;;;;;;;;;;;;;CAcX,CAAA;AAqCD,eAAe,MAAM,CAAA"}