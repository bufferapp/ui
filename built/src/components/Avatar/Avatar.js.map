{"version":3,"file":"Avatar.js","sourceRoot":"","sources":["../../../../src/components/Avatar/Avatar.tsx"],"names":[],"mappings":"AACA,OAAO,KAAK,MAAM,OAAO,CAAA;AAEzB,OAAO,SAAS,MAAM,YAAY,CAAA;AAClC,OAAO,MAAM,MAAM,mBAAmB,CAAA;AACtC,OAAO,KAAK,MAAM,MAAM,SAAS,CAAA;AACjC,OAAO,EACL,QAAQ,EACR,cAAc,EACd,SAAS,EACT,QAAQ,EACR,SAAS,EACT,OAAO,EACP,MAAM,EACN,OAAO,EACP,SAAS,EACT,QAAQ,EACR,OAAO,GACR,MAAM,iBAAiB,CAAA;AACxB,OAAO,EACL,QAAQ,EACR,GAAG,EACH,SAAS,EACT,QAAQ,EACR,SAAS,EACT,OAAO,EACP,MAAM,EACN,OAAO,EACP,SAAS,EACT,QAAQ,EACR,OAAO,GACR,MAAM,SAAS,CAAA;AAEhB,MAAM,OAAO,GAAG,MAAM,CAAC,GAAG,CAAA;IACtB,CAAC,KAAK,EAAE,EAAE,CAEb,MAAM,CAAC,OAAO,CAAC,KAAK,CAAC,IAAI,CAAC;CAC1B,CAAA;AAED,MAAM,KAAK,GAAG,MAAM,CAAC,GAAG,CAAA;IACpB,CAAC,KAAK,EAAE,EAAE,CAEb,MAAM,CAAC,KAAK,CAAC,KAAK,CAAC,IAAI,CAAC,CAAC,KAAK,CAAC,IAAI,CAAC;CACpC,CAAA;AAED,MAAM,aAAa,GAAG,IAAI,GAAG,CAAC;IAC5B;QACE,WAAW;QAEX,CAAC,EAAE,IAAI,EAAE,EAAE,EAAE,CAAC,CACZ,oBAAC,MAAM,CAAC,iBAAiB,IAAC,IAAI,EAAE,IAAI,EAAE,OAAO,EAAE,SAAS;YACtD,oBAAC,SAAS,IAAC,IAAI,EAAC,OAAO,EAAC,KAAK,EAAC,OAAO,GAAG,CACf,CAC5B;KACF;IACD;QACE,UAAU;QAEV,CAAC,EAAE,IAAI,EAAE,EAAE,EAAE,CAAC,CACZ,oBAAC,MAAM,CAAC,iBAAiB,IAAC,IAAI,EAAE,IAAI,EAAE,OAAO,EAAE,QAAQ;YACrD,oBAAC,QAAQ,IAAC,IAAI,EAAC,OAAO,EAAC,KAAK,EAAC,OAAO,GAAG,CACd,CAC5B;KACF;IACD;QACE,SAAS;QAET,CAAC,EAAE,IAAI,EAAE,EAAE,EAAE,CAAC,CACZ,oBAAC,MAAM,CAAC,iBAAiB,IAAC,IAAI,EAAE,IAAI,EAAE,OAAO,EAAE,OAAO;YACpD,oBAAC,OAAO,IAAC,IAAI,EAAC,OAAO,EAAC,KAAK,EAAC,OAAO,GAAG,CACb,CAC5B;KACF;IACD;QACE,UAAU;QAEV,CAAC,EAAE,IAAI,EAAE,EAAE,EAAE,CAAC,CACZ,oBAAC,MAAM,CAAC,iBAAiB,IAAC,IAAI,EAAE,IAAI,EAAE,OAAO,EAAE,QAAQ;YACrD,oBAAC,QAAQ,IAAC,IAAI,EAAC,OAAO,EAAC,KAAK,EAAC,OAAO,GAAG,CACd,CAC5B;KACF;IACD;QACE,WAAW;QAEX,CAAC,EAAE,IAAI,EAAE,EAAE,EAAE,CAAC,CACZ,oBAAC,MAAM,CAAC,iBAAiB,IAAC,IAAI,EAAE,IAAI,EAAE,OAAO,EAAE,SAAS;YACtD,oBAAC,SAAS,IAAC,IAAI,EAAC,OAAO,EAAC,KAAK,EAAC,OAAO,GAAG,CACf,CAC5B;KACF;IACD;QACE,gBAAgB;QAEhB,CAAC,EAAE,IAAI,EAAE,EAAE,EAAE,CAAC,CACZ,oBAAC,MAAM,CAAC,iBAAiB,IAAC,IAAI,EAAE,IAAI,EAAE,OAAO,EAAE,cAAc;YAC3D,oBAAC,GAAG,IAAC,IAAI,EAAC,OAAO,EAAC,KAAK,EAAC,OAAO,GAAG,CACT,CAC5B;KACF;IACD;QACE,SAAS;QAET,CAAC,EAAE,IAAI,EAAE,EAAE,EAAE,CAAC,CACZ,oBAAC,MAAM,CAAC,iBAAiB,IAAC,IAAI,EAAE,IAAI,EAAE,OAAO,EAAE,OAAO;YACpD,oBAAC,OAAO,IAAC,IAAI,EAAC,OAAO,EAAC,KAAK,EAAC,OAAO,GAAG,CACb,CAC5B;KACF;IACD;QACE,QAAQ;QAER,CAAC,EAAE,IAAI,EAAE,EAAE,EAAE,CAAC,CACZ,oBAAC,MAAM,CAAC,iBAAiB,IAAC,IAAI,EAAE,IAAI,EAAE,OAAO,EAAE,MAAM;YACnD,oBAAC,MAAM,IAAC,IAAI,EAAC,OAAO,EAAC,KAAK,EAAC,OAAO,GAAG,CACZ,CAC5B;KACF;IACD;QACE,WAAW;QAEX,CAAC,EAAE,IAAI,EAAE,EAAE,EAAE,CAAC,CACZ,oBAAC,MAAM,CAAC,iBAAiB,IAAC,IAAI,EAAE,IAAI,EAAE,OAAO,EAAE,SAAS;YACtD,oBAAC,MAAM,CAAC,aAAa;gBACnB,oBAAC,SAAS,IAAC,IAAI,EAAC,OAAO,EAAC,KAAK,EAAC,OAAO,GAAG;gBAAC,GAAG,CACvB,CACE,CAC5B;KACF;IACD;QACE,UAAU;QAEV,CAAC,EAAE,IAAI,EAAE,EAAE,EAAE,CAAC,CACZ,oBAAC,MAAM,CAAC,iBAAiB,IAAC,IAAI,EAAE,IAAI,EAAE,OAAO,EAAE,QAAQ;YACrD,oBAAC,QAAQ,IAAC,IAAI,EAAC,OAAO,EAAC,KAAK,EAAC,OAAO,GAAG,CACd,CAC5B;KACF;IACD;QACE,SAAS;QAET,CAAC,EAAE,IAAI,EAAE,EAAE,EAAE,CAAC,CACZ,oBAAC,MAAM,CAAC,iBAAiB,IAAC,IAAI,EAAE,IAAI,EAAE,OAAO,EAAE,OAAO;YACpD,oBAAC,OAAO,IAAC,IAAI,EAAC,OAAO,EAAC,KAAK,EAAC,OAAO,GAAG,CACb,CAC5B;KACF;CACF,CAAC,CAAA;AAGF,MAAM,MAAM,GAAG,CAAC,EAAE,GAAG,EAAE,GAAG,EAAE,IAAI,EAAE,IAAI,EAAE,OAAO,EAAE,WAAW,EAAE,EAAE,EAAE;IAChE,MAAM,UAAU,GAAG,OAAO,IAAI,aAAa,CAAC,GAAG,CAAC,OAAO,CAAC,CAAA;IACxD,OAAO,CACL,oBAAC,OAAO,IAAC,IAAI,EAAE,IAAI;QAChB,UAAU,IAAI,oBAAC,UAAU,IAAC,IAAI,EAAE,IAAI,GAAI;QACzC,oBAAC,KAAK,IACJ,IAAI,EAAE,IAAI,EACV,IAAI,EAAE,IAAI,EACV,GAAG,EAAE,GAAG,EACR,WAAW,EAAE,WAAW,EACxB,GAAG,EAAE,GAAG,GACR,CACM,CACX,CAAA;AACH,CAAC,CAAA;AAED,MAAM,CAAC,SAAS,GAAG;IAEjB,GAAG,EAAE,SAAS,CAAC,MAAM,CAAC,UAAU;IAEhC,GAAG,EAAE,SAAS,CAAC,MAAM,CAAC,UAAU;IAEhC,WAAW,EAAE,SAAS,CAAC,MAAM;IAE7B,IAAI,EAAE,SAAS,CAAC,KAAK,CAAC,CAAC,SAAS,EAAE,QAAQ,EAAE,QAAQ,CAAC,CAAC;IAEtD,IAAI,EAAE,SAAS,CAAC,KAAK,CAAC,CAAC,OAAO,EAAE,QAAQ,EAAE,OAAO,CAAC,CAAC;IAEnD,OAAO,EAAE,SAAS,CAAC,KAAK,CAAC;QACvB,UAAU;QACV,SAAS;QACT,WAAW;QACX,UAAU;QACV,QAAQ;QACR,WAAW;QACX,SAAS;QACT,QAAQ;QACR,WAAW;QACX,gBAAgB;QAChB,UAAU;QACV,SAAS;KACV,CAAC;CACH,CAAA;AAED,MAAM,CAAC,YAAY,GAAG;IACpB,IAAI,EAAE,SAAS;IACf,IAAI,EAAE,OAAO;IACb,WAAW,EAAE,EAAE;IACf,OAAO,EAAE,IAAI;CACd,CAAA;AAED,eAAe,MAAM,CAAA","sourcesContent":["/* eslint-disable */\nimport React from 'react'\n\nimport PropTypes from 'prop-types'\nimport styled from 'styled-components'\nimport * as Styles from './style'\nimport {\n  facebook,\n  googlebusiness,\n  instagram,\n  linkedin,\n  pinterest,\n  shopify,\n  tiktok,\n  twitter,\n  startpage,\n  mastodon,\n  youtube,\n} from '../style/colors'\nimport {\n  Facebook,\n  Gbp,\n  Instagram,\n  LinkedIn,\n  Pinterest,\n  Shopify,\n  Tiktok,\n  Twitter,\n  StartPage,\n  Mastodon,\n  Youtube,\n} from '../Icon'\n\nconst Wrapper = styled.div`\n  ${(props) =>  \n// @ts-expect-error TS(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message\n Styles.wrapper[props.size]}\n`\n\nconst Image = styled.div`\n  ${(props) =>  \n// @ts-expect-error TS(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message\n Styles.image[props.type][props.size]}\n`\n\nconst socialIconMap = new Map([\n  [\n    'instagram',\n    // @ts-expect-error TS(7031) FIXME: Binding element 'size' implicitly has an 'any' typ... Remove this comment to see the full error message\n    ({ size }) => (\n      <Styles.SocialIconWrapper size={size} bgColor={instagram}>\n        <Instagram size=\"small\" color=\"white\" />\n      </Styles.SocialIconWrapper>\n    ),\n  ],\n  [\n    'facebook',\n    // @ts-expect-error TS(7031) FIXME: Binding element 'size' implicitly has an 'any' typ... Remove this comment to see the full error message\n    ({ size }) => (\n      <Styles.SocialIconWrapper size={size} bgColor={facebook}>\n        <Facebook size=\"small\" color=\"white\" />\n      </Styles.SocialIconWrapper>\n    ),\n  ],\n  [\n    'twitter',\n    // @ts-expect-error TS(7031) FIXME: Binding element 'size' implicitly has an 'any' typ... Remove this comment to see the full error message\n    ({ size }) => (\n      <Styles.SocialIconWrapper size={size} bgColor={twitter}>\n        <Twitter size=\"small\" color=\"white\" />\n      </Styles.SocialIconWrapper>\n    ),\n  ],\n  [\n    'linkedin',\n    // @ts-expect-error TS(7031) FIXME: Binding element 'size' implicitly has an 'any' typ... Remove this comment to see the full error message\n    ({ size }) => (\n      <Styles.SocialIconWrapper size={size} bgColor={linkedin}>\n        <LinkedIn size=\"small\" color=\"white\" />\n      </Styles.SocialIconWrapper>\n    ),\n  ],\n  [\n    'pinterest',\n    // @ts-expect-error TS(7031) FIXME: Binding element 'size' implicitly has an 'any' typ... Remove this comment to see the full error message\n    ({ size }) => (\n      <Styles.SocialIconWrapper size={size} bgColor={pinterest}>\n        <Pinterest size=\"small\" color=\"white\" />\n      </Styles.SocialIconWrapper>\n    ),\n  ],\n  [\n    'googlebusiness',\n    // @ts-expect-error TS(7031) FIXME: Binding element 'size' implicitly has an 'any' typ... Remove this comment to see the full error message\n    ({ size }) => (\n      <Styles.SocialIconWrapper size={size} bgColor={googlebusiness}>\n        <Gbp size=\"small\" color=\"white\" />\n      </Styles.SocialIconWrapper>\n    ),\n  ],\n  [\n    'shopify',\n    // @ts-expect-error TS(7031) FIXME: Binding element 'size' implicitly has an 'any' typ... Remove this comment to see the full error message\n    ({ size }) => (\n      <Styles.SocialIconWrapper size={size} bgColor={shopify}>\n        <Shopify size=\"small\" color=\"white\" />\n      </Styles.SocialIconWrapper>\n    ),\n  ],\n  [\n    'tiktok',\n    // @ts-expect-error TS(7031) FIXME: Binding element 'size' implicitly has an 'any' typ... Remove this comment to see the full error message\n    ({ size }) => (\n      <Styles.SocialIconWrapper size={size} bgColor={tiktok}>\n        <Tiktok size=\"small\" color=\"white\" />\n      </Styles.SocialIconWrapper>\n    ),\n  ],\n  [\n    'startPage',\n    // @ts-expect-error TS(7031) FIXME: Binding element 'size' implicitly has an 'any' typ... Remove this comment to see the full error message\n    ({ size }) => (\n      <Styles.SocialIconWrapper size={size} bgColor={startpage}>\n        <Styles.StartPageIcon>\n          <StartPage size=\"small\" color=\"white\" />{' '}\n        </Styles.StartPageIcon>\n      </Styles.SocialIconWrapper>\n    ),\n  ],\n  [\n    'mastodon',\n    // @ts-expect-error TS(7031) FIXME: Binding element 'size' implicitly has an 'any' typ... Remove this comment to see the full error message\n    ({ size }) => (\n      <Styles.SocialIconWrapper size={size} bgColor={mastodon}>\n        <Mastodon size=\"small\" color=\"white\" />\n      </Styles.SocialIconWrapper>\n    ),\n  ],\n  [\n    'youtube',\n    // @ts-expect-error TS(7031) FIXME: Binding element 'size' implicitly has an 'any' typ... Remove this comment to see the full error message\n    ({ size }) => (\n      <Styles.SocialIconWrapper size={size} bgColor={youtube}>\n        <Youtube size=\"small\" color=\"white\" />\n      </Styles.SocialIconWrapper>\n    ),\n  ],\n])\n\n// @ts-expect-error TS(7031) FIXME: Binding element 'src' implicitly has an 'any' type... Remove this comment to see the full error message\nconst Avatar = ({ src, alt, type, size, network, fallbackUrl }) => {\n  const SocialIcon = network && socialIconMap.get(network)\n  return (\n    <Wrapper size={size}>\n      {SocialIcon && <SocialIcon size={size} />}\n      <Image\n        size={size}\n        type={type}\n        src={src}\n        fallbackUrl={fallbackUrl}\n        alt={alt}\n      />\n    </Wrapper>\n  )\n}\n\nAvatar.propTypes = {\n  /** The source of the avatar image. */\n  src: PropTypes.string.isRequired,\n  /** The alt text for the avatar image. */\n  alt: PropTypes.string.isRequired,\n  /** The fallback url for the avatar image. */\n  fallbackUrl: PropTypes.string,\n  /** Can be `'default'` (default, plain avatar), `'status'` (online/offline indicator) or `'social'` (has social network icon from the `network` prop). */\n  type: PropTypes.oneOf(['default', 'social', 'status']),\n  /** Can be `'small'`, `'medium'` or `'large'` (`32px`, `40px` and `48px` respectively). */\n  size: PropTypes.oneOf(['large', 'medium', 'small']),\n  /** Name of social network icon to overlay. (E.g., `'instagram'`). Only applicable when `{ type: 'social' }` */\n  network: PropTypes.oneOf([\n    'facebook',\n    'twitter',\n    'instagram',\n    'linkedin',\n    'google',\n    'pinterest',\n    'shopify',\n    'tiktok',\n    'startPage',\n    'googlebusiness',\n    'mastodon',\n    'youtube',\n  ]),\n}\n\nAvatar.defaultProps = {\n  type: 'default',\n  size: 'small',\n  fallbackUrl: '',\n  network: null,\n}\n\nexport default Avatar\n"]}