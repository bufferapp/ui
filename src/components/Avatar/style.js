import styled, { css } from 'styled-components'
import { yellowDarker } from '../style/colors'
import { fontWeightBold, fontFamily } from '../style/fonts'

const getImageCss = ({ size, type = 'default' }) =>
  css`
    width: ${size}px;
    height: ${size}px;
    border-radius: 100%;
    -webkit-mask-size: cover;
    ${type === 'social' &&
    `-webkit-mask-image: url(https://static.buffer.com/ui/avatar-mask-${size}.svg);`}
    background: url(${(props) => props.src}) center no-repeat, url(${(props) =>
      props.fallbackUrl}) center no-repeat;
    background-size: ${size}px;
  `

export const image = {
  default: {
    small: getImageCss({ size: 32 }),
    medium: getImageCss({ size: 40 }),
    large: getImageCss({ size: 48 }),
  },
  social: {
    small: getImageCss({ size: 32, type: 'social' }),
    medium: getImageCss({ size: 40, type: 'social' }),
    large: getImageCss({ size: 48, type: 'social' }),
  },
}

const getWrapperCss = ({ size }) =>
  css`
    position: relative;
    width: ${size}px;
    height: ${size}px;
  `

export const wrapper = {
  small: getWrapperCss({ size: 32 }),
  medium: getWrapperCss({ size: 40 }),
  large: getWrapperCss({ size: 48 }),
}

const getObjectCss = ({ size }) =>
  css`
    border-radius: 100%;
    width: ${size}px;
    height: ${size}px;
  `

export const object = {
  small: getObjectCss({ size: 32 }),
  medium: getObjectCss({ size: 40 }),
  large: getObjectCss({ size: 48 }),
}

export const SocialIconWrapper = styled.div`
  position: absolute;
  bottom: 0;
  right: 0;
  width: 16px;
  height: 16px;
  border-radius: 100%;
  background: ${(props) => (props.bgColor ? props.bgColor : '#777')};
  display: flex;
  justify-content: center;
  align-items: center;

  transform: translateX(8px)
    translateY(${(props) => (props.size === 'small' ? '0' : '-2')}px);
`

export const StartPageIcon = styled.span`
  color: ${yellowDarker};
  font-size: 9px;
  font-weight: ${fontWeightBold};
  text-align: center;
  width: 16px;
  font-family: ${fontFamily};
`
