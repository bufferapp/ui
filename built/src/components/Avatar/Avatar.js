import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import * as Styles from './style';
import { facebook, googlebusiness, instagram, linkedin, pinterest, shopify, tiktok, twitter, startpage, mastodon, youtube, } from '../style/colors';
import { Facebook, Gbp, Instagram, LinkedIn, Pinterest, Shopify, Tiktok, Twitter, StartPage, Mastodon, Youtube, } from '../Icon';
const Wrapper = styled.div `
  ${(props) => Styles.wrapper[props.size]}
`;
const Image = styled.div `
  ${(props) => Styles.image[props.type][props.size]}
`;
const socialIconMap = new Map([
    [
        'instagram',
        ({ size }) => (React.createElement(Styles.SocialIconWrapper, { size: size, bgColor: instagram },
            React.createElement(Instagram, { size: "small", color: "white" }))),
    ],
    [
        'facebook',
        ({ size }) => (React.createElement(Styles.SocialIconWrapper, { size: size, bgColor: facebook },
            React.createElement(Facebook, { size: "small", color: "white" }))),
    ],
    [
        'twitter',
        ({ size }) => (React.createElement(Styles.SocialIconWrapper, { size: size, bgColor: twitter },
            React.createElement(Twitter, { size: "small", color: "white" }))),
    ],
    [
        'linkedin',
        ({ size }) => (React.createElement(Styles.SocialIconWrapper, { size: size, bgColor: linkedin },
            React.createElement(LinkedIn, { size: "small", color: "white" }))),
    ],
    [
        'pinterest',
        ({ size }) => (React.createElement(Styles.SocialIconWrapper, { size: size, bgColor: pinterest },
            React.createElement(Pinterest, { size: "small", color: "white" }))),
    ],
    [
        'googlebusiness',
        ({ size }) => (React.createElement(Styles.SocialIconWrapper, { size: size, bgColor: googlebusiness },
            React.createElement(Gbp, { size: "small", color: "white" }))),
    ],
    [
        'shopify',
        ({ size }) => (React.createElement(Styles.SocialIconWrapper, { size: size, bgColor: shopify },
            React.createElement(Shopify, { size: "small", color: "white" }))),
    ],
    [
        'tiktok',
        ({ size }) => (React.createElement(Styles.SocialIconWrapper, { size: size, bgColor: tiktok },
            React.createElement(Tiktok, { size: "small", color: "white" }))),
    ],
    [
        'startPage',
        ({ size }) => (React.createElement(Styles.SocialIconWrapper, { size: size, bgColor: startpage },
            React.createElement(Styles.StartPageIcon, null,
                React.createElement(StartPage, { size: "small", color: "white" }),
                ' '))),
    ],
    [
        'mastodon',
        ({ size }) => (React.createElement(Styles.SocialIconWrapper, { size: size, bgColor: mastodon },
            React.createElement(Mastodon, { size: "small", color: "white" }))),
    ],
    [
        'youtube',
        ({ size }) => (React.createElement(Styles.SocialIconWrapper, { size: size, bgColor: youtube },
            React.createElement(Youtube, { size: "small", color: "white" }))),
    ],
]);
const Avatar = ({ src, alt, type, size, network, fallbackUrl }) => {
    const SocialIcon = network && socialIconMap.get(network);
    return (React.createElement(Wrapper, { size: size },
        SocialIcon && React.createElement(SocialIcon, { size: size }),
        React.createElement(Image, { size: size, type: type, src: src, fallbackUrl: fallbackUrl, alt: alt })));
};
Avatar.propTypes = {
    src: PropTypes.string.isRequired,
    alt: PropTypes.string.isRequired,
    fallbackUrl: PropTypes.string,
    type: PropTypes.oneOf(['default', 'social', 'status']),
    size: PropTypes.oneOf(['large', 'medium', 'small']),
    network: PropTypes.oneOf([
        'facebook',
        'twitter',
        'instagram',
        'linkedin',
        'google',
        'pinterest',
        'shopify',
        'tiktok',
        'startPage',
        'googlebusiness',
        'mastodon',
        'youtube',
    ]),
};
Avatar.defaultProps = {
    type: 'default',
    size: 'small',
    fallbackUrl: '',
    network: null,
};
export default Avatar;
//# sourceMappingURL=Avatar.js.map