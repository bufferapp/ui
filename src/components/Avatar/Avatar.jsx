/* eslint-disable */
import React from 'react'

import PropTypes from 'prop-types'
import styled from 'styled-components'
import * as Styles from './style'
import {
  facebook,
  googlebusiness,
  instagram,
  linkedin,
  pinterest,
  shopify,
  tiktok,
  twitter,
  startpage,
  mastodon,
  youtube,
} from '../style/colors'
import {
  Facebook,
  Gbp,
  Instagram,
  LinkedIn,
  Pinterest,
  Shopify,
  Tiktok,
  Twitter,
  StartPage,
  Mastodon,
  Youtube,
} from '../Icon'

const Wrapper = styled.div`
  ${(props) => Styles.wrapper[props.size]}
`

const Image = styled.div`
  ${(props) => Styles.image[props.type][props.size]}
`

const socialIconMap = new Map([
  [
    'instagram',
    ({ size }) => (
      <Styles.SocialIconWrapper size={size} bgColor={instagram}>
        <Instagram size="small" color="white" />
      </Styles.SocialIconWrapper>
    ),
  ],
  [
    'facebook',
    ({ size }) => (
      <Styles.SocialIconWrapper size={size} bgColor={facebook}>
        <Facebook size="small" color="white" />
      </Styles.SocialIconWrapper>
    ),
  ],
  [
    'twitter',
    ({ size }) => (
      <Styles.SocialIconWrapper size={size} bgColor={twitter}>
        <Twitter size="small" color="white" />
      </Styles.SocialIconWrapper>
    ),
  ],
  [
    'linkedin',
    ({ size }) => (
      <Styles.SocialIconWrapper size={size} bgColor={linkedin}>
        <LinkedIn size="small" color="white" />
      </Styles.SocialIconWrapper>
    ),
  ],
  [
    'pinterest',
    ({ size }) => (
      <Styles.SocialIconWrapper size={size} bgColor={pinterest}>
        <Pinterest size="small" color="white" />
      </Styles.SocialIconWrapper>
    ),
  ],
  [
    'googlebusiness',
    ({ size }) => (
      <Styles.SocialIconWrapper size={size} bgColor={googlebusiness}>
        <Gbp size="small" color="white" />
      </Styles.SocialIconWrapper>
    ),
  ],
  [
    'shopify',
    ({ size }) => (
      <Styles.SocialIconWrapper size={size} bgColor={shopify}>
        <Shopify size="small" color="white" />
      </Styles.SocialIconWrapper>
    ),
  ],
  [
    'tiktok',
    ({ size }) => (
      <Styles.SocialIconWrapper size={size} bgColor={tiktok}>
        <Tiktok size="small" color="white" />
      </Styles.SocialIconWrapper>
    ),
  ],
  [
    'startPage',
    ({ size }) => (
      <Styles.SocialIconWrapper size={size} bgColor={startpage}>
        <Styles.StartPageIcon>
          <StartPage size="small" color="white" />{' '}
        </Styles.StartPageIcon>
      </Styles.SocialIconWrapper>
    ),
  ],
  [
    'mastodon',
    ({ size }) => (
      <Styles.SocialIconWrapper size={size} bgColor={mastodon}>
        <Mastodon size="small" color="white" />
      </Styles.SocialIconWrapper>
    ),
  ],
  [
    'youtube',
    ({ size }) => (
      <Styles.SocialIconWrapper size={size} bgColor={youtube}>
        <Youtube size="small" color="white" />
      </Styles.SocialIconWrapper>
    ),
  ],
])

const Avatar = ({ src, alt, type, size, network, fallbackUrl }) => {
  const SocialIcon = network && socialIconMap.get(network)
  return (
    <Wrapper size={size}>
      {SocialIcon && <SocialIcon size={size} />}
      <Image
        size={size}
        type={type}
        src={src}
        fallbackUrl={fallbackUrl}
        alt={alt}
      />
    </Wrapper>
  )
}

Avatar.propTypes = {
  /** The source of the avatar image. */
  src: PropTypes.string.isRequired,
  /** The alt text for the avatar image. */
  alt: PropTypes.string.isRequired,
  /** The fallback url for the avatar image. */
  fallbackUrl: PropTypes.string,
  /** Can be `'default'` (default, plain avatar), `'status'` (online/offline indicator) or `'social'` (has social network icon from the `network` prop). */
  type: PropTypes.oneOf(['default', 'social', 'status']),
  /** Can be `'small'`, `'medium'` or `'large'` (`32px`, `40px` and `48px` respectively). */
  size: PropTypes.oneOf(['large', 'medium', 'small']),
  /** Name of social network icon to overlay. (E.g., `'instagram'`). Only applicable when `{ type: 'social' }` */
  network: PropTypes.oneOf([
    'facebook',
    'twitter',
    'instagram',
    'linkedin',
    'google',
    'pinterest',
    'shopify',
    'tiktok',
    'startPage',
    'googlebusiness',
    'mastodon',
    'youtube',
  ]),
}

Avatar.defaultProps = {
  type: 'default',
  size: 'small',
  fallbackUrl: '',
  network: null,
}

export default Avatar
