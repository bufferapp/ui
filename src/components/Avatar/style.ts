import styled, { css } from 'styled-components'
import { yellowDarker } from '../style/colors'
import { fontWeightBold, fontFamily } from '../style/fonts'

// @ts-expect-error TS(7031) FIXME: Binding element 'size' implicitly has an 'any' typ... Remove this comment to see the full error message
const getImageCss = ({ size, type = 'default' }) =>
  css`
    width: ${size}px;
    height: ${size}px;
    border-radius: 100%;
    -webkit-mask-size: cover;
    ${type === 'social' &&
    `-webkit-mask-image: url(https://static.buffer.com/ui/avatar-mask-${size}.svg);`}
    background: url(${(props) =>
      // @ts-expect-error TS(2339) FIXME: Property 'src' does not exist on type 'ThemeProps<... Remove this comment to see the full error message
      props.src}) center no-repeat, url(${(props) =>
      // @ts-expect-error TS(2339) FIXME: Property 'fallbackUrl' does not exist on type 'The... Remove this comment to see the full error message
      props.fallbackUrl}) center no-repeat;
    background-size: ${size}px;
  `

export const image = {
  default: {
    small: getImageCss({ size: 32 }),
    medium: getImageCss({ size: 40 }),
    large: getImageCss({ size: 48 }),
  },
  social: {
    small: getImageCss({ size: 32, type: 'social' }),
    medium: getImageCss({ size: 40, type: 'social' }),
    large: getImageCss({ size: 48, type: 'social' }),
  },
}

// @ts-expect-error TS(7031) FIXME: Binding element 'size' implicitly has an 'any' typ... Remove this comment to see the full error message
const getWrapperCss = ({ size }) =>
  css`
    position: relative;
    width: ${size}px;
    height: ${size}px;
  `

export const wrapper = {
  small: getWrapperCss({ size: 32 }),
  medium: getWrapperCss({ size: 40 }),
  large: getWrapperCss({ size: 48 }),
}

// @ts-expect-error TS(7031) FIXME: Binding element 'size' implicitly has an 'any' typ... Remove this comment to see the full error message
const getObjectCss = ({ size }) =>
  css`
    border-radius: 100%;
    width: ${size}px;
    height: ${size}px;
  `

export const object = {
  small: getObjectCss({ size: 32 }),
  medium: getObjectCss({ size: 40 }),
  large: getObjectCss({ size: 48 }),
}

export const SocialIconWrapper = styled.div`
  position: absolute;
  bottom: 0;
  right: 0;
  width: 16px;
  height: 16px;
  border-radius: 100%;
  background: ${(props): string => (props.bgColor ? props.bgColor : '#777')};
  display: flex;
  justify-content: center;
  align-items: center;

  transform: translateX(8px)
    translateY(${(props): string => (props.size === 'small' ? '0' : '-2')}px);
`

export const StartPageIcon = styled.span`
  color: ${yellowDarker};
  font-size: 9px;
  font-weight: ${fontWeightBold};
  text-align: center;
  width: 16px;
  font-family: ${fontFamily};
`
