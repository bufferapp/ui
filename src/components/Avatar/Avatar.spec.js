/* eslint-disable react/jsx-filename-extension */
import snap from 'jest-auto-snapshots'
import 'jest-styled-components'

import Avatar from './Avatar'

snap(Avatar, './Avatar.jsx')
