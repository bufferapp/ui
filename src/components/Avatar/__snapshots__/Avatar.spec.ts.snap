// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > Avatar Matches snapshot when passed all props 1`] = `
<styled.div
  size="large"
>
  <Unknown
    size="large"
  />
  <styled.div
    alt="jest-auto-snapshots String Fixture"
    fallbackUrl="jest-auto-snapshots String Fixture"
    size="large"
    src="jest-auto-snapshots String Fixture"
    type="default"
  />
</styled.div>
`;

exports[`jest-auto-snapshots > Avatar Matches snapshot when passed only required props 1`] = `
<styled.div
  size="small"
>
  <styled.div
    alt="jest-auto-snapshots String Fixture"
    fallbackUrl=""
    size="small"
    src="jest-auto-snapshots String Fixture"
    type="default"
  />
</styled.div>
`;
